cloudify_hostpool/storage/sqlite.py

```diff
@@ -0,0 +1,123 @@
+# #######
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+import json
+import sqlite3
+from base import AbstractStorage
+
+
+class SQLiteStorage(AbstractStorage):
+    """ Storage wrapper for SQLite DB implementing AbstractStorage interface"""
+    _CREATE_TABLE = 'CREATE TABLE IF NOT EXISTS servers ' \
+                    '(global_id integer PRIMARY KEY, server_id text, ' \
+                    'address text, auth text, port integer, alive integer, ' \
+                    'reserved integer)'
+
+    def __init__(self, db_filename):
+        self._filename = db_filename
+        self._create_table()
+
+    def _create_table(self):
+        with sqlite3.connect(self._filename) as conn:
+            cursor = conn.cursor()
+            cursor.execute(SQLiteStorage._CREATE_TABLE)
+
+    def _dict_factory(self, cursor, row):
+        """ Create dictionary out of fetched row by db cursor"""
+        d = {}
+        for idx, col in enumerate(cursor.description):
+            if col[0] == 'auth':
+                # "auth" column value is dictionary serialized to string
+                # -> json.dumps(auth).
+                d[col[0]] = json.loads(row[idx])
+            else:
+                d[col[0]] = row[idx]
+        return d
+
+    def _get_sql_and_values_from_kwargs(self, **kwargs):
+        """ Helper method to create sql query """
+        values = tuple(kwargs.itervalues())
+        sql_part = ' AND '.join('{0}=?'.format(k) for k in kwargs)
+        return sql_part, values
+
+    def get_servers(self, **kwargs):
```

Comment:
In fact I so got used to the previous version that I haven't even notice that change. I guess naming filters is more semantic correctly but kwargs is easier for a developer to read. But I've changed to your version.